Demo/src/main/java/com/braintreepayments/demo/ShopperInsightsFragment.kt

```diff
@@ -31,142 +53,224 @@ class ShopperInsightsFragment : BaseFragment() {
+        super.onViewCreated(view, savedInstanceState)
+        initializeViews(view)
+
+        actionButton.setOnClickListener { fetchShopperInsights() }
+        venmoButton.setOnClickListener { launchVenmo() }
+        payPalVaultButton.setOnClickListener { launchPayPalVault() }
+    }
+
+    private fun initializeViews(view: View) {
+        responseTextView = view.findViewById(R.id.responseTextView)
+        actionButton = view.findViewById(R.id.actionButton)
+        payPalVaultButton = view.findViewById(R.id.payPalVaultButton)
+        venmoButton = view.findViewById(R.id.venmoButton)
+        emailInput = view.findViewById(R.id.emailInput)
+        countryCodeInput = view.findViewById(R.id.countryCodeInput)
+        nationalNumberInput = view.findViewById(R.id.nationalNumberInput)
+        emailNullSwitch = view.findViewById(R.id.emailNullSwitch)
+        phoneNullSwitch = view.findViewById(R.id.phoneNullSwitch)
+
+        emailInput.editText?.setText("PR1_merchantname@personal.example.com")
+        nationalNumberInput.editText?.setText("4082321001")
+        countryCodeInput.editText?.setText("1")
+    }
+
+    override fun onResume() {
+        super.onResume()
+        handleVenmoReturnToApp()
+        handlePayPalReturnToApp()
+    }
+
+    private fun handlePayPalReturnToApp() {
+        if (this::paypalPendingRequest.isInitialized) {
+            when (val request = paypalPendingRequest) {
+                is PayPalPendingRequest.Started -> {
+                    val paypalPaymentAuthResult =
+                        paypalLauncher.handleReturnToAppFromBrowser(request, Intent())
+                    if (paypalPaymentAuthResult is PayPalPaymentAuthResult.Success) {
+                        payPalClient.tokenize(paypalPaymentAuthResult) {
+                            when (it) {
+                                is PayPalResult.Success -> {
+                                    val action =
+                                        ShopperInsightsFragmentDirections
+                                            .actionShopperInsightsFragmentToDisplayNonceFragment(
+                                            it.nonce
+                                        )
+                                    NavHostFragment.findNavController(this).navigate(action)
+                                }
+
+                                is PayPalResult.Failure -> {}
+                                is PayPalResult.Cancel -> {}
+                            }
+                        }
+                    } else {
+                        // No PayPal result i.e. not auth'd in
+                        Toast.makeText(
+                            requireContext(),
+                            "Failed to authenticate user.",
+                            Toast.LENGTH_LONG
+                        )
+                            .show()
+                    }
+                }
+
+                is PayPalPendingRequest.Failure -> {}
```

Comment:
Same question here and below on Venmo, not sure if we should be handling these states in some way